File: Skornel02.ETR.Common/Dtos/ExamDto.cs
using Skornel02.ETR.Common.Enums;

namespace Skornel02.ETR.Common.Dtos;

public record ExamDto(
    string CourseCode,
    string CourseSemester,
    DateTimeOffset Start,
    DateTimeOffset End,
    ExamType ExamType,
    string ClassRoomAddress,
    string ClassRoomNumber,
    string ClassRoomName
);

File: Skornel02.ETR.Web/src/schemas/ExamDto.ts
import { z } from 'zod';
import { ExamType } from '../enums/examtypes';

export const ExamSchema = z.object({
	courseCode: z.string(),
	courseSemester: z.string(),
	examType: z.nativeEnum(ExamType),
	start: z.string(),
	end: z.string(),
	classRoomAddress: z.string(),
	classRoomNumber: z.string(),
	classRoomName: z.string()
});

export type Exam = z.infer<typeof ExamSchema>;

export const ExamDtoSchema = z.array(ExamSchema);

export type ExamDto = z.infer<typeof ExamDtoSchema>;
